core/store/src/lib.rs

```diff
@@ -332,6 +333,22 @@ pub fn read_with_cache<'a, T: BorshDeserialize + 'a>(
     Ok(None)
 }
 
+pub fn read_with_cell_cache<'a, T: BorshDeserialize + Clone + 'a>(
+    storage: &Store,
+    col: DBCol,
+    cache: &'a CellLruCache<Vec<u8>, T>,
+    key: &[u8],
+) -> io::Result<Option<T>> {
+    if let Some(value) = cache.get(key) {
+        return Ok(Some(value));
+    }
+    if let Some(result) = storage.get_ser(col, key)? {
+        cache.put(key.to_vec(), result);
+        return Ok(cache.get(key));
```

Comment:
or could we just return Ok(Some(result)) ?

Reply:
Right, good catch, removing this double lookup actually was one of the initiall motivations here :sweat_smile: